ArduPlane/Plane.h

```diff
@@ -785,6 +785,10 @@ class Plane : public AP_Vehicle {
     // soaring mode-change timer
     uint32_t soaring_mode_timer;
 
+    // terrain disable for non AUTO modes, set with an RC Option switch
```

Comment:
You've used a general term here, but specified fbwb and cruise elsewhere.

If we can create a callback on the flightmode to determine whether the RC disablement is valid for the mode that would be better than calling the modes out individually.  This is what we do a lot of in Copter - e.g. `has_manual_throttle`

Reply:
callbacks created in mode.h....retested in SITL...should be ready to go now